.github/actions/valgrind_run/action.yml

```diff
@@ -0,0 +1,11 @@
+name: 'Pyccel valgrind script'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Test with valgrind for memory leaks
+      run: |
+        pyccel --language=c leaks_check.py
+        valgrind --leak-check=full --error-exitcode=1 ./prog_leaks_check
```

Comment:
Shouldn't we be compiling in debug mode to use valgrind? I think that means the command should be:
```
pyccel --language=c --flags="-g" leaks_check.py
```

Reply:
You are right. Moreover, we should probably turn off any optimization by passing the `-O0` flag.